### PR TITLE
[web] Migrate index.html to use flutter.js

### PR DESCRIPTION
**Pull request by ditman**
We have just landed `flutter.js` in Flutter master channel, which greatly simplifies the `index.html` file, and adds a JS API that lets programmers initialize their apps however they see fit. See [this](https://github.com/flutter/flutter/pull/100177).

As a "demo" of the new flexibility, I created a small loading animation with 3 steps:

* "breathing", 50% opacity: main.dart.js is being downloaded.
* "breathing", 100% opacity: canvaskit and other assets needed by the framework being downloaded.
* "zooooom": when the app is ready to start (we delayed this a few ms so people can perceive the zoom effect!).

I've deployed a demo here:

* https://dit-tests.web.app (remember to empty your cache, or throttle your network to better see the different steps!)

Tests:

Manually tested in Chrome/Firefox on Linux and MacOS. (Couldn't find tests for the index.html file?)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by ditman** [comment]
(I am not a designer + this is tuneable, if there's any interest to land it)

File: web/index.html [hunk 1]
```diff
@@ -33,74 +33,66 @@
...
+            resolve(appRunner.runApp());
+          }, 300);
+        });
+      }).then(function(app) {
+        loading.remove();
       });
-    } else {
-      // Service workers not supported. Just drop the <script> tag.
-      loadMainDartJs();
-    }
```
Comment (ditman) [inline web/index.html:103]: If we don't want the `loading` div behavior, the Flutter init script can be simplified to:

```js
_flutter.loader.loadEntrypoint({
  serviceWorker: {
    serviceWorkerVersion: serviceWorkerVersion,
  }
}).then(function(engine) {
  engine.autoStart();
});
```

File: web/index.html [hunk 2]
```diff
@@ -33,74 +33,66 @@
...
-              console.log('New service worker available.');
-              reg.update();
-              waitForActivation(reg.installing);
-            } else {
-              // Existing service worker is still good.
-              console.log('Loading app from service worker.');
-              loadMainDartJs();
-            }
-          });
+  <div id="loading">
```
Comment (ditman) [inline web/index.html:44]: This bit is optional, implements the loading animation.

**Comment by guidezpl** [comment]
Thanks for this, looks great! The gallery is effectively community owned so these sorts of improvements are welcome! The gallery does run off the master channel

**Review by guidezpl — APPROVED** [review]

File: web/index.html [hunk 3]
```diff
@@ -33,74 +33,66 @@
...
+        }
+      }).then(function(engineInitializer) {
+        loading.classList.add('main_done');
+        return engineInitializer.initializeEngine();
+      }).then(function(appRunner) {
+        loading.classList.add('init_done');
+        return new Promise(function(resolve, _) {
+          window.setTimeout(function() {
+            resolve(appRunner.runApp());
+          }, 300);
```
[the remark this answers is not in the document]
Comment (ditman) [reply web/index.html:89]: Ha! I had it documented, and then removed it. I'll add it back :P This is just so people have "time" to see the "zooom" animation :P

[the remark this answers is not in the document]
Comment (ditman) [reply web/index.html]: I changed this slightly, so we don't delay runApp, but actually the point where we delete the loading indicator (it'll either get covered by the flutter app, or will be removed after a big % of the zoom animation has been visible)

**Comment by ditman** [comment]
@guidezpl shall I click "Squash and merge" or do you want to take another look? I think the latest revision is clearer (and it doesn't delay the start of the app!)

Thanks!

**Comment by guidezpl** [comment]
LGTM!

**Pull request merged**